From ddbd4007b7150bafa0f4ec7a5b4862e319a293f7 Mon Sep 17 00:00:00 2001
From: coypu <coypu@sdf.org>
Date: Mon, 8 Feb 2016 00:06:06 +0200
Subject: [PATCH] gdate: Suppress string format literal warning

Newer versions of GCC emit an error here, but we know it's safe.
https://bugzilla.gnome.org/761550

Signed-off-by: Bert van Hall <bert.vanhall@avionic-design.de>

gdate: Move warning pragma outside of function

Commit 0817af40e8c74c721c30f6ef482b1f53d12044c7 breaks the build on
older versions of GCC, which don't allow pragma inside functions.

https://bugzilla.gnome.org/761550

Signed-off-by: Bert van Hall <bert.vanhall@avionic-design.de>
---
 glib/glib/gdate.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/glib/glib/gdate.c b/glib/glib/gdate.c
index 1978cf7..20e6c4a 100644
--- a/glib/glib/gdate.c
+++ b/glib/glib/gdate.c
@@ -2439,6 +2439,9 @@ win32_strftime_helper (const GDate     *d,
  *
  * Returns: number of characters written to the buffer, or 0 the buffer was too small
  */
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wformat-nonliteral"
+
 gsize     
 g_date_strftime (gchar       *s, 
                  gsize        slen, 
@@ -2549,3 +2552,5 @@ g_date_strftime (gchar       *s,
   return retval;
 #endif
 }
+
+#pragma GCC diagnostic pop
-- 
2.7.4

